tests/foreman/api/test_ansible.py

```diff
@@ -253,3 +257,65 @@ def test_add_and_remove_ansible_role_hostgroup(target_sat):
         target_sat.api.HostGroup(id=hg.id).remove_ansible_role(data={'ansible_role_id': role})
     host_roles = target_sat.api.HostGroup(id=hg.id).list_ansible_roles()
     assert len(host_roles) == 0
+
+
+@pytest.mark.rhel_ver_match('[78]')
+@pytest.mark.tier2
+def test_positive_read_facts_with_filter(target_sat, module_org, module_location, rex_contenthost):
+    """
+    Read host's Ansible facts as a user with a role that has host filter
+
+    :id: 483d5faf-7a4c-4cb7-b14f-369768ad99b0
+
+    :Steps:
+        1. Create a role with a host view filtered
+        2. Create a user with that role
+        3. Create a host
+        4. Run Ansible roles on a host
+        5. Using API, read Ansible facts of that host
+
+    :expectedresults: Ansible facts returned
+
+    :BZ: 1699188
+
+    :customerscenario: true
+    """
+    api = target_sat.api
+
+    role = api.Role(
+        name=gen_string('alpha'), location=[module_location], organization=[module_org]
+    ).create()
+    # assign view_hosts (with a filter, to test BZ 1699188),
+    # view_hostgroups, view_facts permissions to the role
+    permission_hosts = api.Permission().search(query={'search': 'name="view_hosts"'})
+    permission_hostgroups = api.Permission().search(query={'search': 'name="view_hostgroups"'})
+    permission_facts = api.Permission().search(query={'search': 'name="view_facts"'})
+    api.Filter(permission=permission_hosts, search='name != nonexistent', role=role).create()
+    api.Filter(permission=permission_hostgroups, role=role).create()
+    api.Filter(permission=permission_facts, role=role).create()
+
+    password = gen_string('alpha')
+    user = api.User(
+        role=[role], password=password, location=[module_location], organization=[module_org]
+    ).create()
+
+    host = api.Host().search(query={'search': f'name={rex_contenthost.hostname}'})[0]
+    host_id = host.id
+    api.Host(id=host_id, organization=[module_org.id]).update(['organization'])
+    api.Host(id=host_id, location=module_location.id).update(['location'])
```

Comment:
imho all lines until here (including) are a part of a test setup, thus i'd move them away to a fixture (of an appt scope). So when it fails, it is distinguishable, that it is the environment, not the test scenario itself.

Also, this test is parameterized, so multiple instances of it would run - in case these would be carried out by the same worker and the fixture would be properly scoped, it could reuse the same `Role`, `Filter` and `User` entities

Apparently, the 2nd paragraph applies to a fixture that would not include defining of the `Host` entities

Reply:
Remade